test/lib/ansible_test/_util/controller/sanity/validate-modules/validate_modules/main.py

```diff
@@ -1260,6 +1260,31 @@ def _validate_ansible_module_call(self, docs):
 
         self._validate_argument_spec(docs, spec, kwargs)
 
+        if isinstance(docs, Mapping) and isinstance(docs.get('attributes'), Mapping):
+            if isinstance(docs['attributes'].get('check_mode'), Mapping):
+                support_value = docs['attributes']['check_mode'].get('support')
+                if not kwargs.get('supports_check_mode', False):
+                    if support_value != 'none':
+                        self.reporter.error(
+                            path=self.object_path,
+                            code='attributes-check-mode',
+                            msg="The module does not declare support for check mode, but the check_mode attribute's"
+                                " support value is '%s' and not 'none'" % support_value
+                        )
+                else:
+                    if support_value not in ('full', 'partial', 'N/A'):
+                        self.reporter.error(
+                            path=self.object_path,
+                            code='attributes-check-mode',
+                            msg="The module does declare support for check mode, but the check_mode attribute's support value is '%s'" % support_value
+                        )
+                if support_value in ('partial', 'N/A') and docs['attributes']['check_mode'].get('details') in (None, '', []):
+                    self.reporter.error(
+                        path=self.object_path,
+                        code='attributes-check-mode-details',
+                        msg="The module declares not fully support for check mode, but has no details on what exactly that means"
```

Comment:
nit:
```suggestion
                        msg="The module declares less than full support for check mode, but has no details on what exactly that means"
```
alternative wording:
```suggestion
                        msg="The module declares it does not fully support check mode, but has no details on what exactly that means"
```